src/Semver.php

```diff
@@ -45,23 +45,15 @@ public static function satisfies($version, $constraints)
     /**
      * Return all versions that satisfy given constraints.
      *
-     * @param string $constraints
      * @param array $versions
+     * @param string $constraints
      *
      * @return array
      */
-    public static function satisfiedBy($constraints, array $versions)
+    public static function satisfiedBy(array $versions, $constraints)
     {
-        if (null === self::$versionParser) {
-            self::$versionParser = new VersionParser();
-        }
-
-        $versionParser = self::$versionParser;
-        $constraints = $versionParser->parseConstraints($constraints);
-        $versions = array_filter($versions, function ($version) use ($constraints, $versionParser) {
-            $version = new Constraint('==', $versionParser->normalize($version));
-
-            return $constraints->matches($version);
+        $versions = array_filter($versions, function ($version) use ($constraints) {
+            return self::satisfies($version, $constraints);
```

Comment:
Ah no you can call as instance if you have an instance, but it's in a static method already, so you just need to replace self:: with the class name I think